### PR TITLE
Aftershock PrepNet Phyle Faction, mobs, locations

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
SUMMARY: Mods " Adding a bionic prepper faction to Aftershock"

#### Purpose of change

Creating a core faction for Aftershock will push it further into it's sci-fi setting and begin to set the tone for what types of additional content belongs in the mod.

#### Describe the solution
Created a faction, 2 monsters, 2 maps, one class and one NPC so far.  

#### Describe alternatives you've considered
Doing less difficult things with my life.

#### Testing
Adding new monsters is the devil.  Literally I may be starting to believe I am a sinner in the hands of an angry god.

#### Additional context
None

### CODE REVIEW

**Comment by Maleclypse** [comment]
@LovamKicsiGazsii I'm getting a load stopping error saying that my npc class is unknown  `` DEBUG    : Error: line 64:33: unknown npc class

        { "monster": "GROUP_PrepNet", "x": [ 3, 20 ], "y": [ 16, 20 ], "repeat": 4 }
      ],
      "place_npcs": [ { "class":
                                ^
                                 "afs_prepnet_gardener", "x": 10, "y": 9 } ]
    }
  


 FUNCTION : bool main_menu::new_character_tab()
 FILE     : src/main_menu.cpp
 LINE     : 741``

I thought you might have an idea where I've gone awry based on your work on the Church faction.  :)

File: data/mods/Aftershock/modinfo.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "ident": "aftershock",
     "name": "Aftershock",
     "authors": [ "Esther", "Coolthulu" ],
-    "maintainers": [ "Esther" ],
+    "maintainers": [ "Esther", "Maleclypse" ],
```
Comment (anothersimulacrum) [inline data/mods/Aftershock/modinfo.json]: They aren't around anymore.
```suggestion
    "maintainers": [ "Maleclypse" ],
```

Comment (Maleclypse) [reply data/mods/Aftershock/modinfo.json]: I wasn’t sure exactly how long they’d been gone so I didn’t want to be premature in removing them. I’ll go ahead and remove.

**Comment by Moonsabrt** [comment]
Atomic churner?
Fascinating :O

**Comment by Maleclypse** [comment]
I'm not sure what to do about the most recent failed CI Build test.

**Comment by Maleclypse** [comment]
> Atomic churner?
> Fascinating :O

Hi-tech pioneer living. :)

**Comment by Maleclypse** [comment]
Doing some testing.  I've listed the cybermastiffs as part of a Mon_faction that is friendly with the NPC faction.  If I attack the NPC the mob doesn't attack me.  If I attack the mob the NPC rushes to my rescue and kills the mob he's supposed to be friendly with.  Any suggestions?

File: data/mods/Aftershock/npcs/afs_prepnet_dialogue.json [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+      { "text": "How do I join the phyle?", "topic": "TALK_prepnet_gardener_ask_membership" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_prepnet_gardener_firstmeet",
+    "dynamic_line": "I'm a gardener for PrepNet Phyle.  Who are you?",
+    "responses": [
+      { "text": "I am a beggar who really needs something to eat.", "topic": "TALK_prepnet_gardener_ask_food" },
+      { "text": "I am just a fellow survivor.", "topic": "TALK_prepnet_gardener_1" },
```
Comment (mlangsdorf) [inline data/mods/Aftershock/npcs/afs_prepnet_dialogue.json]:
```suggestion
      { "text": "I am just a fellow survivor.", "topic": "TALK_PrepNet_gardener_1" },
```

File: data/mods/Aftershock/npcs/afs_prepnet_dialogue.json [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+      { "text": "I can pay.", "topic": "TALK_prepnet_gardener_ask_food_bribe" },
+      { "text": "What do you need done?", "topic": "TALK_MISSION_LIST" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_prepnet_gardener_ask_food_bribe",
+    "dynamic_line": "You have to earn the right to trade with us. Plus we created our own currency before the cataclysm",
+    "responses": [
+      { "text": "It was worth a try.", "topic": "TALK_prepnet_gardener_1" },
```
Comment (mlangsdorf) [inline data/mods/Aftershock/npcs/afs_prepnet_dialogue.json]:
```suggestion
      { "text": "It was worth a try.", "topic": "TALK_PrepNet_gardener_1" },
```

File: data/mods/Aftershock/npcs/afs_prepnet_dialogue.json [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+      { "text": "What kind of currency?", "topic": "TALK_prepnet_gardener_currency" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_prepnet_gardener_ask_membership",
+    "dynamic_line": "This is just an outpost, you'd have to find our main camp and currently we don't want you to.",
+    "responses": [ { "text": "I didn't want to join your club anyway.", "topic": "TALK_prepnet_gardener_1" } ]
```
Comment (mlangsdorf) [inline data/mods/Aftershock/npcs/afs_prepnet_dialogue.json]:
```suggestion
    "responses": [ { "text": "I didn't want to join your club anyway.", "topic": "TALK_PrepNet_gardener_1" } ]
```

File: data/mods/Aftershock/npcs/afs_prepnet_dialogue.json [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    "responses": [
+      { "text": "Ok.", "topic": "TALK_prepnet_gardener_1" },
+      { "text": "What are crypto coins?", "topic": "TALK_prepnet_gardener_currency" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_prepnet_gardener_ask_past",
+    "dynamic_line": "I got introduced to the movement through Mutual Aid Society. From there it became an all consuming part of my life to prepare for a post consumption world. It seems to have worked better for us than for many others.",
+    "responses": [ { "text": "Let's talk about something else.", "topic": "TALK_prepnet_gardener_1" } ]
```
Comment (mlangsdorf) [inline data/mods/Aftershock/npcs/afs_prepnet_dialogue.json]:
```suggestion
    "responses": [ { "text": "Let's talk about something else.", "topic": "TALK_PrepNet_gardener_1" } ]
```

File: data/mods/Aftershock/npcs/afs_prepnet_dialogue.json [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_prepnet_gardener_ask_mood",
+    "dynamic_line": "It can get lonely out here between caravans.  But there's plenty of work and my Cyber Mastiffs.  If a horde comes through we hunker down in a nearby shelter if it's too large for us.",
+    "responses": [ { "text": "I'm glad you stay busy.", "topic": "TALK_prepnet_gardener_1" } ]
```
Comment (mlangsdorf) [inline data/mods/Aftershock/npcs/afs_prepnet_dialogue.json]:
```suggestion
    "responses": [ { "text": "I'm glad you stay busy.", "topic": "TALK_PrepNet_gardener_1" } ]
```

**Comment by mlangsdorf** [comment]
```diff
diff --git a/data/mods/Aftershock/npcs/afs_prepnet_dialogue.json b/data/mods/Aftershock/npcs/afs_prepnet_dialogue.json
index e7de38c681..6b98f13b78 100644
--- a/data/mods/Aftershock/npcs/afs_prepnet_dialogue.json
+++ b/data/mods/Aftershock/npcs/afs_prepnet_dialogue.json
@@ -19,103 +19,103 @@
         "text": "Who are you?",
         "effect": { "u_add_var": "u_met_prepnet_gardener", "type": "general", "context": "meeting", "value": "yes" },
         "condition": { "not": { "u_has_var": "u_met_prepnet_gardener", "type": "general", "context": "meeting", "value": "yes" } },
-        "topic": "TALK_prepnet_gardener_firstmeet"
+        "topic": "TALK_PrepNet_gardener_firstmeet"
       },
       {
         "text": "What is this place?",
         "condition": { "u_has_var": "u_met_prepnet_gardener", "type": "general", "context": "meeting", "value": "yes" },
-        "topic": "TALK_prepnet_gardener_place"
+        "topic": "TALK_PrepNet_gardener_place"
       },
       {
         "text": "What's your story?",
         "condition": { "u_has_var": "u_met_prepnet_gardener", "type": "general", "context": "meeting", "value": "yes" },
-        "topic": "TALK_prepnet_gardener_ask_past"
+        "topic": "TALK_PrepNet_gardener_ask_past"
       },
       {
         "text": "How are things here?",
         "condition": { "u_has_var": "u_met_prepnet_gardener", "type": "general", "context": "meeting", "value": "yes" },
-        "topic": "TALK_prepnet_gardener_ask_mood"
+        "topic": "TALK_PrepNet_gardener_ask_mood"
       },
       {
         "text": "Can I do anything for you?",
         "condition": { "u_has_var": "u_met_prepnet_gardener", "type": "general", "context": "meeting", "value": "yes" },
         "topic": "TALK_MISSION_LIST"
       },
-      { "text": "How do I join the phyle?", "topic": "TALK_prepnet_gardener_ask_membership" }
+      { "text": "How do I join the phyle?", "topic": "TALK_PrepNet_gardener_ask_membership" }
     ]
   },
   {
     "type": "talk_topic",
-    "id": "TALK_prepnet_gardener_firstmeet",
+    "id": "TALK_PrepNet_gardener_firstmeet",
     "dynamic_line": "I'm a gardener for PrepNet Phyle.  Who are you?",
     "responses": [
-      { "text": "I am a beggar who really needs something to eat.", "topic": "TALK_prepnet_gardener_ask_food" },
-      { "text": "I am just a fellow survivor.", "topic": "TALK_prepnet_gardener_1" },
+      { "text": "I am a beggar who really needs something to eat.", "topic": "TALK_PrepNet_gardener_ask_food" },
+      { "text": "I am just a fellow survivor.", "topic": "TALK_PrepNet_gardener_1" },
       { "text": "I'm bringing order back to the land.", "topic": "TALK_PrepNet_gardener_1" }
     ]
   },
   {
     "type": "talk_topic",
-    "id": "TALK_prepnet_gardener_place",
+    "id": "TALK_PrepNet_gardener_place",
     "dynamic_line": "This is a PrepNet Phyle orchard. We were a community of survivalists gathering resources to prepare for climate change, but it left us better prepared for what actually happened than most.",
     "responses": [
-      { "text": "Can you give me something to eat then?", "topic": "TALK_prepnet_gardener_ask_food" },
-      { "text": "What can I buy here?", "topic": "TALK_prepnet_gardener_ask_goods" },
-      { "text": "Oh, okay.", "topic": "TALK_prepnet_gardener_1" }
+      { "text": "Can you give me something to eat then?", "topic": "TALK_PrepNet_gardener_ask_food" },
+      { "text": "What can I buy here?", "topic": "TALK_PrepNet_gardener_ask_goods" },
+      { "text": "Oh, okay.", "topic": "TALK_PrepNet_gardener_1" }
     ]
   },
   {
     "type": "talk_topic",
-    "id": "TALK_prepnet_gardener_food",
+    "id": "TALK_PrepNet_gardener_ask_food",
     "dynamic_line": "This food has to go back to our core communities. But maybe we can trade some food for services",
     "responses": [
-      { "text": "I can pay.", "topic": "TALK_prepnet_gardener_ask_food_bribe" },
+      { "text": "I can pay.", "topic": "TALK_PrepNet_gardener_ask_food_bribe" },
       { "text": "What do you need done?", "topic": "TALK_MISSION_LIST" }
     ]
   },
   {
     "type": "talk_topic",
-    "id": "TALK_prepnet_gardener_ask_food_bribe",
+    "id": "TALK_PrepNet_gardener_ask_food_bribe",
     "dynamic_line": "You have to earn the right to trade with us. Plus we created our own currency before the cataclysm",
     "responses": [
-      { "text": "It was worth a try.", "topic": "TALK_prepnet_gardener_1" },
-      { "text": "What kind of currency?", "topic": "TALK_prepnet_gardener_currency" }
+      { "text": "It was worth a try.", "topic": "TALK_PrepNet_gardener_1" },
+      { "text": "What kind of currency?", "topic": "TALK_PrepNet_gardener_currency" }
     ]
   },
   {
     "type": "talk_topic",
-    "id": "TALK_prepnet_gardener_ask_membership",
+    "id": "TALK_PrepNet_gardener_ask_membership",
     "dynamic_line": "This is just an outpost, you'd have to find our main camp and currently we don't want you to.",
-    "responses": [ { "text": "I didn't want to join your club anyway.", "topic": "TALK_prepnet_gardener_1" } ]
+    "responses": [ { "text": "I didn't want to join your club anyway.", "topic": "TALK_PrepNet_gardener_1" } ]
   },
   {
     "type": "talk_topic",
-    "id": "TALK_prepnet_gardener_ask_goods",
+    "id": "TALK_PrepNet_gardener_ask_goods",
     "dynamic_line": "Here? Fruits and berries. Maybe the occasional piece of farm equipment, but you need crypto coins",
     "responses": [
-      { "text": "Ok.", "topic": "TALK_prepnet_gardener_1" },
-      { "text": "What are crypto coins?", "topic": "TALK_prepnet_gardener_currency" }
+      { "text": "Ok.", "topic": "TALK_PrepNet_gardener_1" },
+      { "text": "What are crypto coins?", "topic": "TALK_PrepNet_gardener_currency" }
     ]
   },
   {
     "type": "talk_topic",
-    "id": "TALK_prepnet_gardener_ask_past",
+    "id": "TALK_PrepNet_gardener_ask_past",
     "dynamic_line": "I got introduced to the movement through Mutual Aid Society. From there it became an all consuming part of my life to prepare for a post consumption world. It seems to have worked better for us than for many others.",
-    "responses": [ { "text": "Let's talk about something else.", "topic": "TALK_prepnet_gardener_1" } ]
+    "responses": [ { "text": "Let's talk about something else.", "topic": "TALK_PrepNet_gardener_1" } ]
   },
   {
     "type": "talk_topic",
-    "id": "TALK_prepnet_gardener_ask_mood",
+    "id": "TALK_PrepNet_gardener_ask_mood",
     "dynamic_line": "It can get lonely out here between caravans.  But there's plenty of work and my Cyber Mastiffs.  If a horde comes through we hunker down in a nearby shelter if it's too large for us.",
-    "responses": [ { "text": "I'm glad you stay busy.", "topic": "TALK_prepnet_gardener_1" } ]
+    "responses": [ { "text": "I'm glad you stay busy.", "topic": "TALK_PrepNet_gardener_1" } ]
   },
   {
     "type": "talk_topic",
-    "id": "TALK_prepnet_gardener_currency",
+    "id": "TALK_PrepNet_gardener_currency",
     "dynamic_line": "We use crypto coins for money here. They were a pre cataclysm attempt to abstract money even further and create electronic cash. We figured we'd keep it up. Barter only gets you so far.",
     "responses": [
-      { "text": "Oh, okay.", "topic": "TALK_prepnet_gardener_1" },
-      { "text": "What can I buy here?", "topic": "TALK_prepnet_gardener_ask_goods" }
+      { "text": "Oh, okay.", "topic": "TALK_PrepNet_gardener_1" },
+      { "text": "What can I buy here?", "topic": "TALK_PrepNet_gardener_ask_goods" }
     ]
   },
   {
```
These are all the changes that need to be made to make all the topics start with `"TALK_PrepNet_gardener"` and for all the topics to be correctly referenced.

**Comment by Maleclypse** [comment]
Faction Fail


I'm trying to make the cyber-mastiff's treat PrepNet_Phyle faction NPCs the same way that Giant Bees treat Apis.  Currently the mastiff's remain neutral until they see zombies.  It appears that once zombies appear they attack everyone including the faction NPCs they are supposed to be aligned with.  I'm not having luck determining how Apis stays neutral to bees in similar situations.

**Comment by Amariithynar** [comment]
The Tripods are unceasingly hostile to me just for coming near to them when they're attacking a Fungal Tower; they stopped prioritizing attacking the Mycus and turned on me. If it wasn't night time, I'd have been mobbed by like 15 of them; as soon as I got away, they completely tore through every fungal creature and structure there, and then started hunting for me. I'm neutral with PrepNet; they should probably not be aggro to me when I'm not infected or Mycus myself.

**Comment by Maleclypse** [comment]
> The Tripods are unceasingly hostile to me just for coming near to them when they're attacking a Fungal Tower; they stopped prioritizing attacking the Mycus and turned on me. If it wasn't night time, I'd have been mobbed by like 15 of them; as soon as I got away, they completely tore through every fungal creature and structure there, and then started hunting for me. I'm neutral with PrepNet; they should probably not be aggro to me when I'm not infected or Mycus myself.

That should be fixed in the upcoming experimental.

**Pull request merged**